packages/utils/src/metrics.ts

```diff
@@ -0,0 +1,80 @@
+export type LabelsGeneric = Record<string, string | number>;
+export type LabelKeys<Labels extends LabelsGeneric> = Extract<keyof Labels, string>;
+export type CollectFn<Labels extends LabelsGeneric> = (metric: Gauge<Labels>) => void;
+
+// Type `{}` is required here as alternative types do not work
+// eslint-disable-next-line @typescript-eslint/ban-types
+export type NoLabels = {};
+
+export interface Gauge<Labels extends LabelsGeneric = NoLabels> {
+  startTimer(): NoLabels extends Labels ? () => number : (labels: Labels) => number;
+  startTimer(labels: NoLabels extends Labels ? undefined : Labels): (labels?: Labels) => number;
+
+  inc: NoLabels extends Labels ? (value?: number) => void : (labels: Labels, value?: number) => void;
+  dec: NoLabels extends Labels ? (value?: number) => void : (labels: Labels, value?: number) => void;
+  set: NoLabels extends Labels ? (value: number) => void : (labels: Labels, value: number) => void;
+
+  collect?(): void;
```

Comment:
Overriding collect is the way to go if you use default prom-client Gauge. If we use the custom GaugeExtra on the other hand we should use addCollect instead. It would be good if we move away from GaugeExtra in the long-term and only use prom-client default metric classes.

I completely removed the collect method from GaugeExtra now as there is no reason to use it there.

see commit https://github.com/ChainSafe/lodestar/pull/6201/commits/2e7d70b0f4d3fa333be38a2cb8d9cfc1948d66f8